### DISCUSSION TITLE
pyodbc 4.0.32 installation getting failed in airflow 2.4.0 base docker image

### DISCUSSION THREAD

**Discussion by Amey2400** [apache/airflow#29510] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

The pyodbc 4.0.32 package installation fails when we try to install it in apache/airflow:2.4.0-python3.9. Pyodbc package has dependency on unixodbc-dev package. 
So recently after building the docker image with apache/airflow:2.4.0-python3.9 as base unixodbc-dev version got updated to 2.3.11. previous to this build the version of unixodbc-dev was 2.3.7. After this update of unixodbc-dev to 2.3.11 we are getting following error during installation of pyodbc 4.0.32 package.
Error message:
`Building wheels for collected packages: pyodbc
  Building wheel for pyodbc (setup.py): started
  Building wheel for pyodbc (setup.py): finished with status 'error'
[91m  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      running bdist_wheel
      running build
      running build_ext
      building 'pyodbc' extension
      creating build
      creating build/temp.linux-x86_64-3.9
      creating build/temp.linux-x86_64-3.9/src
      gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYODBC_VERSION=4.0.32 -I/usr/local/include/python3.9 -c src/buffer.cpp -o build/temp.linux-x86_64-3.9/src/buffer.o -Wno-write-strings
      In file included from /usr/include/sql.h:19,
                       from src/pyodbc.h:56,
                       from src/buffer.cpp:12:
      /usr/include/sqltypes.h:56:10: fatal error: unixodbc.h: No such file or directory
         56 | #include "unixodbc.h"
            |          ^~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
[0m[91m  ERROR: Failed building wheel for pyodbc
[0m  Running setup.py clean for pyodbc
Failed to build pyodbc
`

### What you think should happen instead

The pyodbc 4.0.32 package should be installed succesfully in docker image having airflow base image without any issue.

### How to reproduce

Create any docker image with apache/airflow:2.4.0-python3.9 as base image and try to install pyodbc 4.0.32 in the same.

### Operating System

Linux-debian

### Deployment

Other Docker-based deployment

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Amey2400** [comment]
Recently one issue in microsoft is raised for unixodbc-2.3.11 for the same issue missing unixodbc.h file. 
https://github.com/microsoft/linux-package-repositories/issues/36

**Reply by potiuk** [reply]
This is not Airlfow/ image - problem so converting it to a discussion.

Just to make it clear - airflow issues are when airflow has a problem and you want to report it, discussions are when you have a problem and need help, so I encourage to use the latter.  Generally speaking you need to figure out how to compile and install such packages - and I guess it's a question of googling how others have done that. Like it is witha any other package, airlfow image  does not have guarantees that installing any version package will be "smooth" - sometimes even some packages will be broken after upgrade. Sometimes you will need to downgrade some package. upgrade them, install new ones - depends what you want to install and which version. Each case is different so it is hard to help to solve your problem always. 

And usually it's best to raise such question with the particular library/package you want to install as it is more likely someone had similar problem. Often (this is what  I do ) since airflow image is just plain debian image with some packages installed, there are usualky some other issues (and solutions) raised elsewhere. So you can usually google the solution.

IMHO (looking at past experience you need to instal unixodbc-dev package in specific versio or similar - this is at leas what first result of my google revealed here:   https://github.com/mkleehammer/pyodbc/issues/441 there are some discussions, approaches and apparently it did solve similar problem so maybe it will work for you - but even if it does  not, I'd encourage you to search for similar issues and solutions.
